x-pack/plugins/stack_connectors/server/connector_types/resilient/resilient.ts

```diff
@@ -0,0 +1,347 @@
+  private urls: {
+    incidentTypes: string;
+    incident: string;
+    comment: string;
+    severity: string;
+  };
+
+  constructor(params: ServiceParams<ResilientConfig, ResilientSecrets>) {
+    super(params);
+
+    this.urls = {
+      incidentTypes: `${this.getIncidentFieldsUrl()}/incident_type_ids`,
+      incident: `${this.getOrgUrl()}/incidents`,
+      comment: `${this.getOrgUrl()}/incidents/{inc_id}/comments`,
+      severity: `${this.getIncidentFieldsUrl()}/severity_code`,
+    };
+
+    this.registerSubActions();
+  }
+
+  protected getResponseErrorMessage(error: AxiosError) {
+    if (!error.response?.status) {
+      return i18n.UNKNOWN_API_ERROR;
+    }
+    if (error.response.status === 401) {
+      return i18n.UNAUTHORIZED_API_ERROR;
+    }
+    return `API Error: ${error.response?.statusText}`;
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.INCIDENT_TYPES,
+      method: 'getIncidentTypes',
+      schema: ExecutorSubActionGetIncidentTypesParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.SEVERITY,
+      method: 'getSeverity',
+      schema: ExecutorSubActionGetSeverityParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.FIELDS,
+      method: 'getFields',
+      schema: ExecutorSubActionCommonFieldsParamsSchema,
+    });
+  }
+
+  private getAuthHeaders() {
+    const token = Buffer.from(
+      this.secrets.apiKeyId + ':' + this.secrets.apiKeySecret,
+      'utf8'
+    ).toString('base64');
+
+    return { Authorization: `Basic ${token}` };
+  }
+
+  private getOrgUrl() {
+    const { apiUrl: url, orgId } = this.config;
+
+    return `${url}/rest/orgs/${orgId}`;
+  }
+
+  private getIncidentFieldsUrl = () => `${this.getOrgUrl()}/types/incident/fields`;
+
+  private getIncidentViewURL(key: string) {
+    const url = this.config.apiUrl;
+    const urlWithoutTrailingSlash = url.endsWith('/') ? url.slice(0, -1) : url;
+
+    return `${urlWithoutTrailingSlash}/${VIEW_INCIDENT_URL}/${key}`;
+  }
+
+  public async createIncident({
+    incident,
+  }: Record<string, unknown>): Promise<ExternalServiceIncidentResponse> {
+    try {
+      const incidentData = incident as Incident;
+      if (!incidentData?.name) {
+        throw new Error('Incident name is required');
+      }
+
+      let data: CreateIncidentData = {
+        name: incidentData.name,
+        discovered_date: Date.now(),
+      };
+
+      if (incidentData?.description) {
+        data = {
+          ...data,
+          description: {
+            format: 'html',
+            content: incidentData.description ?? '',
+          },
+        };
+      }
+
+      if (incidentData?.incidentTypes) {
+        data = {
+          ...data,
+          incident_type_ids: incidentData.incidentTypes.map((id: number | string) => ({
+            id: Number(id),
+          })),
+        };
+      }
+
+      if (incidentData?.severityCode) {
```

Comment:
I find it strange that the severity is not mandatory but IBM Resilient allows it 🤷 

The UI is a bit weird though. When configuring a rule the default value is empty but once we pick a value we cannot select empty again. Wdyt?

https://github.com/elastic/kibana/assets/1533137/ec8b8a43-f1de-43cc-9782-4fdb7ca5a734

Reply:
yup, this behaviour is also common with other connectors like Jira, ServiceNow etc. 
But I am not sure it should to be changed or not.